### PR TITLE
Implement device path discovery for NVMe support

### PR DESCRIPTION
**Pull request by leakingtapan**
**Is this a bug fix or adding new feature?**
Add support for nvme on Nitro instance
Fixes: #178 

**What testing is done?** 
Manually tested following scenario:
 - [x] nvme on m5.xlarge with filesystem mode
 - [x] nvme on m5.xlarge with block volume mode

### CODE REVIEW

File: pkg/driver/node.go [hunk 1]
```diff
@@ -89,11 +90,18 @@ func (d *Driver) NodeStageVolume(ctx context.Context, req *csi.NodeStageVolumeRe
...
 
-	source, ok := req.PublishContext[DevicePathKey]
+	devicePath, ok := req.PublishContext[DevicePathKey]
 	if !ok {
 		return nil, status.Error(codes.InvalidArgument, "Device path not provided")
 	}
 
+	source, err := d.findDevicePath(devicePath, volumeID)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Device path %s does not exist. %v", devicePath, err)
```
Comment (dkoshkin) [inline pkg/driver/node.go]: This error is misleading, it's not always that it doesn't exist, there could be other errors.

File: pkg/driver/node.go [hunk 2]
```diff
@@ -359,3 +379,56 @@ func (d *Driver) nodePublishVolumeForFileSystem(req *csi.NodePublishVolumeReques
 
 	return nil
 }
+
+// findDevicePath finds and verifies the path of device
+// if the device is not nvme device, return the path
+// if the device is nvme device, returns the actual device path
```
Comment (dkoshkin) [inline pkg/driver/node.go]: What does "actual device path" mean vs "the path of device"?

File: pkg/driver/node.go [hunk 3]
```diff
@@ -359,3 +379,56 @@ func (d *Driver) nodePublishVolumeForFileSystem(req *csi.NodePublishVolumeReques
...
+
+// findNvmeVolume looks for the nvme volume with the specified name
+// It follows the symlink (if it exists) and returns the absolute path to the device
+func findNvmeVolume(findName string) (device string, err error) {
+	p := filepath.Join("/dev/disk/by-id/", findName)
+	stat, err := os.Lstat(p)
+	if err != nil {
+		if os.IsNotExist(err) {
+			klog.V(5).Infof("nvme path %q not found", p)
+			return "", nil
```
Comment (dkoshkin) [inline pkg/driver/node.go]: Should this return an error? The condition of `""` isn't checked in the previous functions.

File: pkg/driver/node.go (hunk 2 above)
Comment (leakingtapan) [reply pkg/driver/node.go]: clarified the docs

**Comment by leakingtapan** [comment]
```
[0m[91mgo: golang.org/x/crypto@v0.0.0-20181203042331-505ab145d0a9: git fetch -f origin refs/heads/*:refs/heads/* refs/tags/*:refs/tags/* in /go/pkg/mod/cache/vcs/de5fd3af413a4f3f969455ae522b4002fcb7bb4c158f339396dfc77710c9007d: exit status 128:
	error: RPC failed; HTTP 502 curl 22 The requested URL returned error: 502 Bad Gateway
	fatal: The remote end hung up unexpectedly
[0m[91mgo: golang.org/x/crypto@v0.0.0-20180904163835-0709b304e793: unknown revision 0709b304e793
[0m[91mgo: golang.org/x/sys@v0.0.0-20181205085412-a5c9d58dba9a: git fetch -f origin refs/heads/*:refs/heads/* refs/tags/*:refs/tags/* in /go/pkg/mod/cache/vcs/76a8992ccba6d77c6bcf031ff2b6d821cf232e4ad8d1f2362404fbd0a798d846: exit status 128:
	error: RPC failed; HTTP 502 curl 22 The requested URL returned error: 502 Bad Gateway
	fatal: The remote end hung up unexpectedly
```

/test pull-aws-ebs-csi-driver-e2e-single-az

**Comment by leakingtapan** [comment]
@dkoshkin if no other concerns, Im going to override the coverage drop and merge the change

**Pull request merged**